### DISCUSSION TITLE
configuring cypress component test with rollup bundler

### DISCUSSION THREAD

**Discussion by DanSallau** [cypress-io/cypress#28570] [Questions and Help]
Please I need help configuring cypress component test with rollup bundler . I have opened a question on stackoverflow [here](https://stackoverflow.com/questions/77699262/cypress-component-test-with-rollup-bundler) and is willing to explain further here . Please any help will be appreciated .

**Comment by portyu9** [comment]
Hi @DanSallau 

You should not need to bundle each Cypress support/spec file manually.

The issue is that serving those files directly from Express is not enough: `.ts`, JSX and their imports need to go through a dev server that performs compilation, module resolution, dependency handling and watching.

Cypress's supported built-in Component Testing dev servers are currently **Vite** and **Webpack**, not Rollup.

If this is a React project, for example, the supported setup is:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  component: {
    devServer: {
      framework: 'react',
      bundler: 'vite',
    },
  },
})
```

Using Vite is usually the natural choice for a project whose production build uses Rollup. Vite uses Rollup in its build pipeline, while also providing the development-time module server Cypress Component Testing needs.

You may need to reproduce relevant aliases/plugins/transforms from your Rollup configuration in Vite's configuration, but you do **not** need to generate a separate Rollup bundle for each Cypress test.

A fully custom Cypress dev server is possible, but then your implementation owns compilation, dependency resolution, watching, serving and shutdown. Unless there is a specific reason that Component Testing must execute through Rollup itself, using Cypress's Vite integration is substantially simpler and is the supported path.

Please mark this answer as accepted if it helped, thank you.
